File: packages/android/src/main/java/com/kozmos/components/POIDetailPanel/POIDetailPanel.figma.kt
package com.kozmos.components.poidetailpanel

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.contracts.KozmosPOIPresentation

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8103")
class KozmosPOIDetailPanelConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Kozmos Cafe"

    @FigmaProperty(FigmaType.Enum, "Presentation")
    val presentation: KozmosPOIDetailPanelPresentation = Figma.mapping(
        "Inline" to KozmosPOIDetailPanelPresentation.Inline,
        "Sheet" to KozmosPOIDetailPanelPresentation.Sheet,
        "Panel" to KozmosPOIDetailPanelPresentation.Panel
    )

    @Composable
    fun ComponentExample() {
        KozmosPOIDetailPanel(
            poi = KozmosPOIPresentation(
                id = "cafe", name = title, floorId = "l2",
                floorLabel = "Level 2", actions = emptyList()
            ),
            actionLabels = emptyMap(),
            onAction = { _, _ -> },
            onClose = {},
            presentation = presentation
        )
    }
}

File: packages/ios/Sources/Components/POIDetailPanel/POIDetailPanel.figma.swift
import SwiftUI
import Figma

struct KozmosPOIDetailPanelConnect: FigmaConnect {
    let component = KozmosPOIDetailPanel.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8103"

    @FigmaString("Title Text")
    var title: String = "Kozmos Cafe"

    @FigmaEnum(
        "Presentation",
        mapping: [
            "Inline": KozmosPOIDetailPanel.Presentation.inline,
            "Sheet": KozmosPOIDetailPanel.Presentation.sheet,
            "Panel": KozmosPOIDetailPanel.Presentation.panel
        ]
    )
    var presentation: KozmosPOIDetailPanel.Presentation = .inline

    var body: some View {
        KozmosPOIDetailPanel(
            poi: KozmosPOIPresentation(
                id: "cafe",
                name: self.title,
                floorId: "l2",
                floorLabel: "Level 2",
                actions: []
            ),
            actionLabels: [:],
            onAction: { _, _ in },
            onClose: {},
            presentation: self.presentation
        )
    }
}
